src/config.rs

```diff
@@ -666,7 +665,7 @@ fn config_file(env_var: &str, leaf: &str) -> PathBuf {
 
 #[derive(Debug, Default, PartialEq, Eq)]
 pub struct Config {
-    pub caches: Vec<CacheType>,
+    pub cache: Option<CacheType>,
```

Comment:
Was this used anywhere before? Did multiple backends ever work?

Reply:
> Did multiple backends ever work?

No, they never works at the same time.

The only case is users configure different backends at the same time, and some of them are wrong. For example, users configure s3, azure, redis at the same time, and s3, azure is invalid, only redis works.

In old logic, the errors returned by s3 and azure will be ignore, and using redis finally. In this PR, we will only choose the first valid cache type and returns error happened which makes the behavior more predictable.

In the future, I will return errors if users configured different backends at the same time.